src/main/java/github/scarsz/discordsrv/modules/alerts/AlertListener.java

```diff
@@ -286,6 +312,96 @@ public void unregister() {
+                try {
+                    // Acknowledge the click to avoid "This interaction failed"
+                    btnEvent.reply("Command queued.").setEphemeral(true).queue();
+                } catch (Throwable ignored) {}
+
+                // Disable the clicked button immediately on the message (non-webhook path only has message objects we can edit)
+                try {
+                    List<ActionRow> current = btnEvent.getMessage().getActionRows();
+                    List<ActionRow> updated = new ArrayList<>(current.size());
+                    boolean changed = false;
+                    for (ActionRow row : current) {
+                        List<Button> newButtons = new ArrayList<>(row.getButtons().size());
+                        for (Button b : row.getButtons()) {
+                            if (id.equals(b.getId())) {
+                                newButtons.add(b.withDisabled(true));
+                                changed = true;
+                            } else {
+                                newButtons.add(b);
+                            }
+                        }
+                        updated.add(ActionRow.of(newButtons));
+                    }
+                    if (changed) {
+                        // Try edit via event helper if available; fallback to editing the message
+                        try {
+                            btnEvent.getMessage().editMessageComponents(updated).queue(s -> {}, e -> {});
+                        } catch (Throwable ignored2) {}
+                    }
+                } catch (Throwable t) {
+                    DiscordSRV.debug(Debug.ALERTS, "Failed to disable clicked button immediately: " + t.getMessage());
+                }
+
+                // Remove mapping to prevent any reuse
+                buttonCommandMap.remove(id);
+
+                // Parse context and command
+                String runContext = "console";
+                String cmd = stored;
+                try {
+                    if (stored.startsWith("console:")) {
+                        runContext = "console";
+                        cmd = stored.substring("console:".length());
+                    } else if (stored.startsWith("player:")) {
+                        runContext = "player";
+                        int first = stored.indexOf(':');
+                        int second = stored.indexOf(':', first + 1);
+                        String playerName = second > first ? stored.substring(first + 1, second) : null;
+                        cmd = second > 0 ? stored.substring(second + 1) : "";
+
+                        final String finalCmd = cmd;
+                        final String finalPlayerName = playerName;
+                        SchedulerUtil.runTask(DiscordSRV.getPlugin(), () -> {
+                            try {
+                                Player p = finalPlayerName != null ? Bukkit.getPlayerExact(finalPlayerName) : null;
```

Comment:
I don't think it's useful to be able to specify player names directly for the command targets. Nobody besides the event player themselves are guaranteed to be online, and even then there's possibilities the player might not be. I'd suggest just changing this to a `player` or `console` switch with `player` automatically invoking an event's #getPlayer() method if it exists. The player lookup can stay as a fallback for events that offer the player names in different methods though